File: backend/module-auth/src/main/java/com/enterprise/auth/dto/PinLoginRequest.java
package com.enterprise.auth.dto;

import jakarta.validation.constraints.AssertTrue;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.UUID;

/**
 * @file PinLoginRequest.java
 * @description POS PIN 碼登入請求 / PIN login request for POS terminal
 * @description_en Accepts either a terminal UUID or a stable terminal code for POS PIN login
 * @description_zh 支援使用終端 UUID 或穩定終端代碼進行 POS PIN 登入
 */
@Data
public class PinLoginRequest {

    @Pattern(regexp = "^\\d{4,6}$", message = "PIN must be 4-6 digits")
    private String pin;

    private UUID terminalId;

    @Size(max = 50, message = "Terminal code must be 50 characters or fewer")
    private String terminalCode;

    @AssertTrue(message = "Terminal ID or terminal code is required")
    public Boolean getTerminalIdentifierPresent() {
        return terminalId != null || (terminalCode != null && !terminalCode.isBlank());
    }
}

File: frontend-app/lib/core/models/pin_login_request.dart
/*
 * @file pin_login_request.dart
 * @description PIN 登入請求模型 / PIN login request model
 * @description_en Captures the POS terminal PIN login payload before HTTP wiring.
 * @description_zh 在 HTTP 串接前定義 POS 終端 PIN 登入請求資料。
 */

class PinLoginRequest {
  const PinLoginRequest({
    required this.pin,
    this.terminalId,
    this.terminalCode,
  });

  final String pin;
  final String? terminalId;
  final String? terminalCode;

  bool get isValid {
    final normalizedPin = pin.trim();
    final hasTerminalId = terminalId != null && terminalId!.trim().isNotEmpty;
    final hasTerminalCode =
        terminalCode != null && terminalCode!.trim().isNotEmpty;

    return RegExp(r'^\d{4,6}$').hasMatch(normalizedPin) &&
        (hasTerminalId || hasTerminalCode);
  }

  Map<String, Object?> toJson() {
    final payload = <String, Object?>{'pin': pin};

    if (terminalId != null && terminalId!.trim().isNotEmpty) {
      payload['terminalId'] = terminalId;
    }

    if (terminalCode != null && terminalCode!.trim().isNotEmpty) {
      payload['terminalCode'] = terminalCode;
    }

    return payload;
  }
}
